src/main/java/org/quantumbadger/redreader/common/General.java

```diff
@@ -401,7 +401,12 @@ public void onClick(DialogInterface dialog, int which) {
 	private static final Pattern urlPattern = Pattern.compile("^(https?)://([^/]+)/+([^\\?#]+)((?:\\?[^#]+)?)((?:#.+)?)$");
 
 	public static String filenameFromString(String url) {
-		return uriFromString(url).getPath().replace(File.separator, "");
+		String filename = uriFromString(url).getPath().replace(File.separator, "");
+		String[] parts = filename.substring(1).split("\\.", 2);
+		if(parts.length < 2)
+			filename += ".jpg";
```

Comment:
In any case, a bad guess is not worse than not guessing.

I would even argue it's better; It won't display properly, but at least it will show up in the gallery.